hera_mc/cm_hookup.py

```diff
@@ -161,31 +187,35 @@ def __get_next_connection(self, direction, part, rev, pol):
                     (func.upper(PC.Connections.down_part_rev) == rev.upper())):
                 conn.gps2Time()
                 if cm_utils._is_active(self.at_date, conn.start_date, conn.stop_date):
-                    nppart = conn.upstream_part
-                    npport = conn.upstream_output_port
                     options.append(copy.copy(conn))
+                    cpo = conn.downstream_input_port
+                    npa = conn.upstream_part
+                    npo = conn.upstream_output_port
         elif direction.lower() == 'down':  # Going downstream
             for conn in self.session.query(PC.Connections).filter(
                     (func.upper(PC.Connections.upstream_part) == part.upper()) &
                     (func.upper(PC.Connections.up_part_rev) == rev.upper())):
                 conn.gps2Time()
                 if cm_utils._is_active(self.at_date, conn.start_date, conn.stop_date):
-                    nppart = conn.downstream_part
-                    npport = conn.downstream_input_port
                     options.append(copy.copy(conn))
+                    cpo = conn.upstream_output_port
+                    npa = conn.downstream_part
+                    npo = conn.downstream_input_port
         next_one = None
         if len(options) == 0:
             next_one = None
         elif len(options) == 1:
-            next_one = options[0]
+            opc = options[0]
+            if self._check_next_port(opc.upstream_part, opc.upstream_output_port, pol, len(options)):
```

Comment:
Does this branch need to check the value of the `direction` variable, like some other spots in this function?

Reply:
Thanks!